From: Carlos Garnacho <carlosg@gnome.org>
Date: Mon, 18 Sep 2023 10:26:47 +0200
Subject: core: Do not repick pointer while syncing wayland foci

This is a remnant of unreliable pointer state after some of the
event grabbing conditions (originally introduced for window dragging
at commit 1b29113150). Since the introduction of ClutterGrab
and implicit grabs generating crossing events when the conditions
change, this is not necessary.

Related: https://gitlab.gnome.org/GNOME/mutter/-/issues/2977
Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/3287>
(cherry picked from commit 39f599018cdc27480db1deea92cc451506a37a5f)

Origin: upstream, after 45.0
---
 src/core/display.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/src/core/display.c b/src/core/display.c
index 3000ab8..0a191c0 100644
--- a/src/core/display.c
+++ b/src/core/display.c
@@ -1480,10 +1480,6 @@ meta_display_sync_wayland_input_focus (MetaDisplay *display)
 #ifdef HAVE_WAYLAND
   MetaWaylandCompositor *compositor = wayland_compositor_from_display (display);
   MetaWindow *focus_window = NULL;
-  MetaBackend *backend = backend_from_display (display);
-  ClutterBackend *clutter_backend = meta_backend_get_clutter_backend (backend);
-  ClutterSeat *seat = clutter_backend_get_default_seat (clutter_backend);
-  MetaStage *stage = META_STAGE (meta_backend_get_stage (backend));
   gboolean is_no_focus_xwindow = FALSE;
 
 #ifdef HAVE_X11_CLIENT
@@ -1503,9 +1499,6 @@ meta_display_sync_wayland_input_focus (MetaDisplay *display)
     meta_topic (META_DEBUG_FOCUS, "Focus change has no effect, because there is no matching wayland surface");
 
   meta_wayland_compositor_set_input_focus (compositor, focus_window);
-
-  clutter_stage_repick_device (CLUTTER_STAGE (stage),
-                               clutter_seat_get_pointer (seat));
 #endif
 }
 
